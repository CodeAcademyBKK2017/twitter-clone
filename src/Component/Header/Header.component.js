import inputStyles from './Header.css';
import React, {Component} from 'react';
import CircleBG from '../Circle/CircleBG.component';
import CircleNoneBG from '../Circle/CircleNoneBG.component';
import InputBorder from '../InputBorder/InputBorder.component';
import Icon from 'react-ionicons';
import className from 'classname';

class Header extends Component {
  state = {
    menuVisibility: false
  }
  toggleMenu =() => {
    this.setState({menuVisibility: !this.state.menuVisibility});
  }
  render () {
    return (
      <div className={className(inputStyles.header, inputStyles.shadow)}>
        { 
          this.state.menuVisibility ?
            <div className={className(inputStyles.sidemenu)}>
              <div className={inputStyles.borderStyle}>
                <input className={inputStyles.inputStyle} placeholder="Search"/>
                <Icon className={inputStyles.colorStyle} icon='ios-search'/>
              </div>
              <div className={inputStyles.name}>Nanny</div>
            </div> : null
        }
        
        <div className={className(inputStyles.show)} onClick={this.toggleMenu}>
          <Icon icon="ios-menu"/></div> 
        <div className={className(inputStyles.flex4, inputStyles.contentLeft)}>
        
          <CircleBG iconName={'logo-twitter'} color='white'/>

          <div className={inputStyles.divide}/>

          <CircleNoneBG color='#5f5f5f' iconName={'ios-notifications'}/>
          <CircleNoneBG color='#5f5f5f' iconName={'ios-mail'}/>
          <CircleNoneBG color='#5f5f5f' iconName={'ios-grid-outline'}/>

        </div>

        <div className={inputStyles.flex1}><Icon fontSize='35px' color='#1e90ff' icon='logo-twitter'/></div>

        <div className={className(inputStyles.flex4, inputStyles.contentRight)}>
          <InputBorder />
          <CircleNoneBG color='red' iconName={'md-heart'}/>
        </div>
        
      </div>

    );
  }
}

export default Header;
