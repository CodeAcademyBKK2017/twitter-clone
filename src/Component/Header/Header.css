  .Box-Header {  
    display: flex;  
    height: 60px;
    position: fixed;
    top:0;
    width:100%;    
    justify-content: center;
    align-items: center;
  } 
  
  .Header4 {
    flex-grow: 4;
  } 
  
  .Header1 {
    flex-grow: 1;
    display: flex;
    justify-content: center;
    align-items: center;
  }

  .logo {
    color:#80A8CC;
    size : 2x;
  }
  
