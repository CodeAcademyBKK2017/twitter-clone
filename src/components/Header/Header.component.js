import HeaderStyle from './Header.style.css';
import LandingPageStyle from '../LandingPage/LandingPage.style.css';
import CircleStyle from '../Circle/Circle.style.css';
import React, {Component} from 'react';
import Circle from '../Circle/Circle.component';
import SearchBar from '../SearchBar/SearchBar.component';
import SideMenu from '../SideMenu/SideMenu.component';
import Icon from 'react-ionicons';

class Header extends Component {

	state = {
		isHideSideMenu: true
	};

	onMenuButtonClick = () => {
		const newState = {...this.state};
		newState.isHideSideMenu = !this.state.isHideSideMenu;
		this.setState(newState);
	}

	render () {
		return (
			<div className={`${LandingPageStyle.box} ${HeaderStyle.twitterHeader}`}>
				<div className={HeaderStyle.mainContainer}>
					<div className={HeaderStyle.leftContent}>
						<div className={HeaderStyle.brandContainer}>
							<Circle isBrand={true} iconName="logo-twitter"/>
						</div>
					
						<div className={HeaderStyle.lineSpliter}/>

						<div className={HeaderStyle.buttonContainer}>
							<Circle iconName="ios-notifications-outline"/>
							<Circle iconName="ios-mail-outline"/>
							<Circle iconName="ios-grid-outline"/>
							<Circle/>
						</div>
					</div>

					<div className={HeaderStyle.logoContent}>
						<Icon className={HeaderStyle.twitterLogo} icon="logo-twitter" fontSize="35px" color="#80A8CC"/>
					</div>

					<div className={HeaderStyle.rightContent}>
						<SearchBar/>

						<div className={HeaderStyle.brandContainer}>
							<Circle>
								<img className={`${CircleStyle.circle} ${CircleStyle.image}`} src="https://cdn1.iconfinder.com/data/icons/ninja-things-1/1772/ninja-simple-512.png" alt=""/>
							</Circle>
						</div>
					</div>
				</div>

				<div className={HeaderStyle.mobileContainer}>
					<div className={HeaderStyle.leftContent}>
						<div className={HeaderStyle.brandContainer}>
							<Circle iconName="ios-menu" onButtonClick={this.onMenuButtonClick}/>
						</div>
					</div>

					<div className={HeaderStyle.logoContent}>
						<Icon className={HeaderStyle.twitterLogo} icon="logo-twitter" fontSize="35px" color="#80A8CC"/>
					</div>

					<div className={HeaderStyle.rightContent} />

					<SideMenu isHidden={this.state.isHideSideMenu}/>
				</div>
			</div>
		);
	}
}

export default Header;
