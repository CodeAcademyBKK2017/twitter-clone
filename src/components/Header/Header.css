.background_white{
    background-color: white;
}

.box_shadow {
    box-shadow: 0 2px 2px 0 rgba(0,0,0,0.08), 0 0 0 1px rgba(0,0,0,0.04);
}

.header {
    position: fixed;
    left: 0;
    right: 0;
    height: 60px;
    top: 0;
    display: flex;
}

.left_head{
    flex: 4;
    display: flex;
    align-items: center;
    justify-content: flex-start;
}
.center_head{
    flex: 1;
    color: #80A8CC;
    display: flex;
    align-items: center;
    justify-content: center;
}
.right_head{
    flex: 4;
    display: flex;
    align-items: center;
    justify-content: flex-end;
}
.right_head_text {
    padding: 5px;
    border: 1px solid gray;
    display: inline-flex;
    border-radius: 10px;
}
.right_head_text > input {
    border: none;
    outline: none;
    background: none;
    width: 20vw;
}
.right_head_icon {
    padding-right: 10px;
}

.head_first_icon {
    padding: 0 10px 0 0;
    border-right: 1px solid black;
    height: 100%;
    align-items: center;
    display: flex;
}
.head_second_icon {
    height: 100%;
    align-items: center;
    display: flex;
}

.header_mobile {
    display: none;
}
.menuBar {
    left: -80vw;
}

@media only screen and (max-width: 768px) {
    .header {
        display: none;
    }
    .header_mobile {
        position: fixed;
        left: 0;
        right: 0;
        height: 60px;
        top: 0;
        display: flex;
    }

    .menuBar{
        /* display: inline-flex; */
        bottom: 0;
        top: 60px;
        position: fixed;
        /* width: 80vw; */
        background-color: #F2F2F2;
        padding: 5px 5px 5px 5px;
        transition: all 0.5s;
    }
    .searchTextBox{
        /* display: inline-flex; */
        width: 100%;
    }
    .right_head_text > input{
        /* display: inline-flex; */
        width: 100%;
    }
    .bottomText{
        position: absolute;
        bottom: 0;
        left: 0;
        right: 0;
        text-align: center;
        padding: 10px;
        border-top: 2px solid #ccc;
    }

    .activeMenu{
        left: 0vw;
        width: 80vw;
    }
}