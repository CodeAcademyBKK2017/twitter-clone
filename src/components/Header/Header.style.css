.header {
    position: fixed;
    left: 0;
    right: 0;
    top: 0;
    bottom: 0;
    height: 60px;
    background-color: white;
    color: black;
    box-shadow: 0 2px 2px 0 rgba(0,0,0,0.08), 0 0 0 1px rgba(0,0,0,0.04);
    display: flex;
}
.left {
    flex: 4;
    display: flex;
    align-items: center;
}
.logo {
    flex: 1;
    height: 100%;
    color: #80A8CC;
    display: flex;
    align-items: center;
    justify-content: center;
}
.right {
    flex: 4;
    display: flex;
    align-items: center;
    justify-content: flex-end;
}
.circleContainer {
    height: 100%;
    display: inline-flex;
    align-items: center;
    justify-content: center;
}
.circleContainer__first {
    width: 55px;
    margin-right: 5px;
}
.circleContainer__user {
    width: 55px;
}
.circleContainer__borderRight {
    border-right: 1px solid #f8f4ee;
}
.circleContainer__borderBottom {
    border-bottom: 2px solid #81a9cb;
}
.circleContainer+.circleContainer {
    width: 45px;
}