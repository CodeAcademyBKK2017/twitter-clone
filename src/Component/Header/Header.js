import './Header.css';
import Circle from '../Circle/Circle'
import React, {Component} from 'react';
import {Icon} from 'react-fa';

class Header extends Component {
  render () {
    return (     
        <div className='Box-Header shadow box-bg'>
          <div className='Header4'><Circle name='twitter'/></div>
          <div className='Header1'><Icon size='2x' className='logo' name="twitter"/></div>
          <div className='Header4'>c</div>        
        </div>      
    );
  }
}

export default Header;
