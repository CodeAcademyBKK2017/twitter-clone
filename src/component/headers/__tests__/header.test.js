import Header from '../header.component';
import React from 'react';
import renderer from 'react-test-renderer';

describe('Header component', () => {
  it('Header: renders correctly', () => {
    const tree = renderer.create(<Header />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});