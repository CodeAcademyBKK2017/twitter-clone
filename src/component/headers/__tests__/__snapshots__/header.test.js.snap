// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component Header: renders correctly 1`] = `
<div
  className="headerTwit shadowStyle"
>
  <div
    className="leftContain isFlex"
  >
    <div
      className="leftConLogo Center isFlex"
    />
    <div
      className="btnGroup isFlex"
    >
      <div
        className="sperateBTN isFlex Center"
      >
        <div
          className="cirBTN isFlex Center"
        />
      </div>
      <div
        className="sperateBTN isFlex Center"
      >
        <div
          className="cirBTN isFlex Center"
        />
      </div>
      <div
        className="sperateBTN isFlex Center"
      >
        <div
          className="cirBTN isFlex Center"
        />
      </div>
    </div>
  </div>
  <div
    className="ConLogo Center isFlex"
  />
  <div
    className="rightContain isFlex"
  >
    <div
      className="sperateBTN isFlex Center"
    >
      <div
        className="cirBTN isFlex Center"
      >
        <img
          className="logoBtn userIcon"
          src="https://scontent-iad3-1.xx.fbcdn.net/v/t39.1997-6/s110x80/13601946_1176746099014660_1320250140_n.png?_nc_ad=z-m&_nc_cid=0&oh=2a7109b26b5725dd0ef7741df62012c4&oe=5AC2F560"
        />
      </div>
    </div>
  </div>
</div>
`;
